packages/fxa-settings/src/models/reliers/oauth-relier.ts

```diff
@@ -376,3 +407,17 @@ function scopeStrToArray(scopes: string) {
     .filter((x) => x.length > 0);
   return new Set(arrScopes);
 }
+
+function replaceItemInArray(
+  array: Array<string>,
+  itemToReplace: string,
+  replaceWith: Array<string>
+) {
+  return [...array.filter((x) => x === itemToReplace), ...replaceWith];
+}
```

Comment:
I'm unclear on what this function is doing. (disclaimer: I'm not very smart)
If I vanilla JS this and throw it into DevTools:
```js
function replaceItemInArray(
  array = [],
  itemToReplace = "",
  replaceWith = []) {
  return [...array.filter((x) => x === itemToReplace), ...replaceWith];
}
```
Then call it with `replaceItemInArray(["one", "two", "three", "four"], "two",  ["peter"]);`
My resulting array is `['two', 'peter']` (I was expecting `['one', 'three', 'four', 'peter']`).

Am I doing something wrong, or is it replacing everything _except_ the `itemToReplace` and then appending my `replaceWith[]` array at the end? Although I can understand the current behavior because the filter is `x === itemToReplace`, but the name doesn't seem to be matching the behavior; but see previous disclaimer.

Reply:
Oh good catch. This was previously done with some lodash and you are right. My translation of this has a bug. The filter should be a `!==` not an `===`. 

btw, this was the initial form:
```
function replaceItemInArray(array, itemToReplace, replaceWith) {
  var without = _.without(array, itemToReplace);
  if (without.length !== array.length) {
    return _.union(without, replaceWith);
  }
  return array;
}
```

Reply:
... btw, I am surprised that the tests I ported still passed with this bug present..

Reply:
Thank you @pdehaan for catching this 😅

I also am surprised the tests still passed.